src/sql/base/browser/ui/table/plugins/tableColumn.ts

```diff
@@ -0,0 +1,173 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { StandardKeyboardEvent } from 'vs/base/browser/keyboardEvent';
+import { Emitter } from 'vs/base/common/event';
+import { KeyCode } from 'vs/base/common/keyCodes';
+
+export interface TableColumn<T extends Slick.SlickData> {
+	readonly definition: Slick.Column<T>;
+}
+
+export interface TableCellClickEventArgs<T extends Slick.SlickData> {
+	item: T;
+	position: { x: number, y: number };
+	row: number;
+	column: number;
+}
+
+export abstract class BaseClickableColumn<T extends Slick.SlickData> implements Slick.Plugin<T>, TableColumn<T> {
```

Comment:
The code here is from the old buttonColumn definition, moved it here for code reuse.